pandas/core/nanops.py

```diff
@@ -457,7 +466,8 @@ def nanmean(values, axis=None, skipna=True, mask=None):
         values, skipna, 0, mask=mask)
     dtype_sum = dtype_max
     dtype_count = np.float64
-    if is_integer_dtype(dtype) or is_timedelta64_dtype(dtype):
+    if (is_integer_dtype(dtype) or is_timedelta64_dtype(dtype) or
```

Comment:
use needs_i8_conversion

Reply:
that includes PeriodDType, which isn't implemented yet (again, #23890 gets these all-at-once by implementing reductions directly on the arrays)